File: frontend/types/runs.ts
/**
 * TypeScript types matching the FastAPI backend Pydantic models for run registry.
 */

export interface RunRegistryItem {
  correlation_id: string
  primitive_name: string
  primitive_version: string
  status: "STARTED" | "SUCCESS" | "FAILED"
  started_at: string // ISO timestamp
  completed_at: string | null // ISO timestamp
  duration_ms: number | null
  input_count: number | null
  decision_count: number | null
  at_risk_count: number | null
  unknown_count: number | null
  error_message: string | null
}

export interface RunRegistryResponse {
  items: RunRegistryItem[]
  next_cursor: string | null
}

File: src/opsiq_runtime/app/api/models/runs.py
"""Pydantic models for run registry API responses."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class RunRegistryItem(BaseModel):
    """Run registry item representing a single runtime run."""

    correlation_id: str
    primitive_name: str
    primitive_version: str
    status: str = Field(..., description="STARTED | SUCCESS | FAILED")
    started_at: datetime = Field(..., description="ISO timestamp")
    completed_at: datetime | None = None
    duration_ms: int | None = None
    input_count: int | None = None
    decision_count: int | None = None
    at_risk_count: int | None = None
    unknown_count: int | None = None
    error_message: str | None = None


class RunRegistryResponse(BaseModel):
    """Response for run registry endpoint."""

    items: list[RunRegistryItem]
    next_cursor: str | None = None

